404 error when paper looks for update

### Describe the problem

Hi
I updated my minecraft server and I get the following error repeating over and over.
I checked and there's no 1.20.1 paper version; only 1.20



```
[init] ERROR: failed to lookup build number for PaperMC version=1.20.1
[init] Running as uid=1000 gid=1000 with /data as 'drwxrwxr-x 1 1000 1000 678 Jun 12 19:54 /data'
[init] Resolved version given LATEST into 1.20.1 and major version 1.20
[init] Resolving type given PAPER
[mc-image-helper] 19:55:42.794 ERROR : 'get' command failed. Version is 1.29.2
java.net.UnknownHostException: papermc.io: Temporary failure in name resolution
        at java.base/java.net.Inet4AddressImpl.lookupAllHostAddr(Native Method)
        at java.base/java.net.InetAddress$PlatformNameService.lookupAllHostAddr(Unknown Source)
        at java.base/java.net.InetAddress.getAddressesFromNameService(Unknown Source)
...
```

I manually updated the paper to 1.20-13.jar but that gives me the same error.
This configuration has worked for a few months; updates were never an issue

### Container definition

```
version: '3.8'
services:
  minecraft:
    image: "itzg/minecraft-server"
    container_name: "minecraft"
    restart: unless-stopped
    ports:
      - "25565:25565"
      - "19132:19132/udp"
      - "19132:19132"
    environment:
      TZ: "Etc/GMT+1"
      EULA: "true"
      TYPE: "PAPER"
      MEMORY: "3G"
      VERSION: "latest"
    tty: true
    stdin_open: true
    volumes:
      - /volume1/docker/minecraft/data:/data:rw
      - /etc/localtime:/etc/localtime:ro
      - /var/run/docker.sock:/var/run/docker.sock #https://github.com/ualex73/monitor_docker

networks:
    default:
        external:
            name: alexandria
```


### Container logs

```
user@host:/volume1/docker/minecraft# docker logs minecraft
[init] Running as uid=1000 gid=1000 with /data as 'drwxrwxr-x 1 1000 1000 678 Jun 12 19:54 /data'
[init] Resolved version given LATEST into 1.20.1 and major version 1.20
[init] Resolving type given PAPER
[mc-image-helper] 19:54:54.225 ERROR : 'get' command failed. Version is 1.29.2
me.itzg.helpers.get.RequestFailedException: Failed to download https://papermc.io/api/v2/projects/paper/versions/1.20.1: status code: 404, reason phrase: Not Found
        at me.itzg.helpers.get.GetCommand.processSingleUri(GetCommand.java:442)
        at me.itzg.helpers.get.GetCommand.call(GetCommand.java:188)
        at me.itzg.helpers.get.GetCommand.call(GetCommand.java:55)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
        at picocli.CommandLine.execute(CommandLine.java:2170)
        at me.itzg.helpers.McImageHelper.main(McImageHelper.java:143)
Caused by: org.apache.hc.client5.http.HttpResponseException: status code: 404, reason phrase: Not Found
        at org.apache.hc.client5.http.impl.classic.AbstractHttpClientResponseHandler.handleResponse(AbstractHttpClientResponseHandler.java:68)
        at me.itzg.helpers.http.LoggingResponseHandler.handleResponse(LoggingResponseHandler.java:20)
        at org.apache.hc.client5.http.impl.classic.CloseableHttpClient.execute(CloseableHttpClient.java:247)
        at org.apache.hc.client5.http.impl.classic.CloseableHttpClient.execute(CloseableHttpClient.java:188)
        at org.apache.hc.client5.http.impl.classic.CloseableHttpClient.execute(CloseableHttpClient.java:162)
        at me.itzg.helpers.get.GetCommand.processSingleUri(GetCommand.java:440)
        ... 11 common frames omitted
[init] ERROR: failed to lookup build number for PaperMC version=1.20.1
[init] Running as uid=1000 gid=1000 with /data as 'drwxrwxr-x 1 1000 1000 678 Jun 12 19:54 /data'
[init] Resolved version given LATEST into 1.20.1 and major version 1.20
[init] Resolving type given PAPER
[mc-image-helper] 19:55:06.541 ERROR : 'get' command failed. Version is 1.29.2
me.itzg.helpers.get.RequestFailedException: Failed to download https://papermc.io/api/v2/projects/paper/versions/1.20.1: status code: 404, reason phrase: Not Found

```

--- a/Dockerfile
+++ b/Dockerfile
@@ -46,7 +46,7 @@ RUN easy-add --var os=${TARGETOS} --var arch=${TARGETARCH}${TARGETVARIANT} \
   --var version=0.1.1 --var app=maven-metadata-release --file {{.app}} \
   --from https://github.com/itzg/{{.app}}/releases/download/{{.version}}/{{.app}}_{{.version}}_{{.os}}_{{.arch}}.tar.gz
 
-ARG MC_HELPER_VERSION=1.29.2
+ARG MC_HELPER_VERSION=1.30.1
 ARG MC_HELPER_BASE_URL=https://github.com/itzg/mc-image-helper/releases/download/${MC_HELPER_VERSION}
 # used for cache busting local copy of mc-image-helper
 ARG MC_HELPER_REV=1

--- a/scripts/start-deployFolia
+++ b/scripts/start-deployFolia
@@ -8,4 +8,4 @@ if [[ $FOLIABUILD ]]; then
   export PAPERBUILD="$FOLIABUILD"
 fi
 
-PAPER_PROJECT="folia" PAPER_NAME="FoliaMC" exec ${SCRIPTS:-/}start-deployPaper "$@"
+PAPER_PROJECT="folia" PAPER_NAME="FoliaMC" exec "${SCRIPTS:-/}start-deployPaper" "$@"

--- a/scripts/start-deployPaper
+++ b/scripts/start-deployPaper
@@ -5,79 +5,48 @@
 set -o pipefail
 handleDebugMode
 
+: "${PAPER_PROJECT:=paper}"
+: "${PAPERBUILD:=}"
+: "${PAPER_DOWNLOAD_URL:=}"
 : "${PAPER_CUSTOM_JAR:=}"
 
-ourScript="$0"
-ourArgs=("$@")
-
-# Check if we're running Folia.
-if [[ -z $PAPER_PROJECT ]]; then
-  PAPER_PROJECT="paper"
-  PAPER_NAME="PaperMC"
-fi
-
-function handleMissingVersion() {
-  expectedVersion=${VANILLA_VERSION}
-  versions=$(curl -fsSL "https://papermc.io/api/v2/projects/${PAPER_PROJECT}" -H  "accept: application/json")
-  if [[ $VERSION = LATEST ]]; then
-    tries=0
-    while ((tries++ < 5)); do
-      VANILLA_VERSION=$(echo "$versions" | jq -r ".versions[$((- tries))]")
-      if [[ $(curl -fsSL "https://papermc.io/api/v2/projects/${PAPER_PROJECT}/versions/${VANILLA_VERSION}" -H  "accept: application/json" \
-               | jq '.builds[-1]') != null ]]; then
-        log "WARN: using ${VANILLA_VERSION} since that's the latest provided by ${PAPER_NAME}"
-        # re-execute the current script with the newly computed version
-        exec "$ourScript" "${ourArgs[@]}"
-     fi
-   done
-  fi
-  log "ERROR: ${expectedVersion} is not published by ${PAPER_NAME}"
-  log "       Set VERSION to one of the following: "
-  log "       $(echo "$versions" | jq -r '.versions | join(", ")')"
-  exit 1
-}
-
+resultsFile=/data/.paper.env
 if [[ $PAPER_CUSTOM_JAR ]]; then
   export SERVER="$PAPER_CUSTOM_JAR"
 elif [[ $PAPER_DOWNLOAD_URL ]]; then
-  SERVER=$(getFilenameFromUrl "${PAPER_DOWNLOAD_URL}")
-  export SERVER
-
-  log "Downloading custom ${PAPER_NAME} jar from $PAPER_DOWNLOAD_URL"
-  if ! mc-image-helper mcopy \
-    --scope=papermc \
-    --to=/data \
-    "${PAPER_DOWNLOAD_URL}"; then
-    echo "ERROR: failed to download ${PAPER_NAME} from $PAPER_DOWNLOAD_URL"
-    exit 1
+  if ! mc-image-helper install-paper \
+    --output-directory=/data \
+    --results-file="$resultsFile" \
+    --url="$PAPER_DOWNLOAD_URL}"; then
+      log "ERROR: failed to download from custom PaperMC URL"
+      exit 1
   fi
+  # grab SERVER and export it
+  set -a
+  # shellcheck disable=SC1090
+  source "${resultsFile}"
+  set +a
 
 else
-  # Paper API v2 docs : https://papermc.io/api/docs/swagger-ui/index.html?configUrl=/api/openapi/swagger-config
-
-  if ! build=${PAPERBUILD:=$(get --json-path=".builds[-1]" "https://papermc.io/api/v2/projects/${PAPER_PROJECT}/versions/${VANILLA_VERSION}")}; then
-    log "ERROR: failed to lookup build number for ${PAPER_NAME} version=${VANILLA_VERSION}"
-    exit 1
+  args=(
+    --output-directory=/data
+    --results-file="$resultsFile"
+    --project="$PAPER_PROJECT"
+    --version="$VERSION"
+  )
+  if [[ $PAPERBUILD ]]; then
+    args+=(--build="$PAPERBUILD")
   fi
-
-  if [[ $build = null ]]; then
-    handleMissingVersion
+  if ! mc-image-helper install-paper "${args[@]}"; then
+      log "ERROR: failed to download $PAPER_PROJECT"
+      exit 1
   fi
+  # grab SERVER and export it
+  set -a
+  # shellcheck disable=SC1090
+  source "${resultsFile}"
+  set +a
 
-  if ! SERVER=$(get --json-path=.downloads.application.name "https://papermc.io/api/v2/projects/${PAPER_PROJECT}/versions/${VANILLA_VERSION}/builds/${build}"); then
-    echo "ERROR: failed to lookup ${PAPER_NAME} download file from version=${VANILLA_VERSION} build=${build}"
-    exit 1
-  fi
-  export SERVER
-
-  log "Downloading ${PAPER_NAME} $VANILLA_VERSION (build $build) ..."
-  if ! mc-image-helper mcopy \
-    --scope=papermc \
-    --to=/data \
-    "https://papermc.io/api/v2/projects/${PAPER_PROJECT}/versions/${VANILLA_VERSION}/builds/${build}/downloads/${SERVER}"; then
-    echo "ERROR: failed to download ${PAPER_NAME} from version=${VANILLA_VERSION} build=${build} download=${SERVER}"
-    exit 1
-  fi
 fi
 
 # Normalize on Spigot for downstream operations

